tools/converter/synapse2ballerina/src/main/java/org/wso2/ei/tools/synapse2ballerina/util/JMSPropertyMapper.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.ei.tools.synapse2ballerina.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Keep Synapse jms property names against ballerina jms property keys
+ */
+public enum JMSPropertyMapper {
+
+    CONTEXT_FACTORY("java.naming.factory.initial", "initialContextFactory"),
+    PROVIDER_URL("java.naming.provider.url", "providerUrl"),
+    CONNECTION_FACTORY("transport.jms.ConnectionFactoryJNDIName", "connectionFactoryName"),
+    DESTINATION_TYPE("transport.jms.DestinationType", "connectionFactoryType");
+
+    JMSPropertyMapper(String propertyName, String ballerinaKey) {
+        this.propertyName = propertyName;
+        this.ballerinaKey = ballerinaKey;
+    }
+
+    private String propertyName;
+    private String ballerinaKey;
+    private static final Map<String, String> ENUM_MAP;
+
+    public String getPropertyName() {
+        return propertyName;
+    }
+
+    public String getBallerinaKey() {
+        return ballerinaKey;
+    }
+
+    public static Map<String, String> getEnumMap() {
+        return ENUM_MAP;
+    }
+
+    static {
+        Map<String, String> map = new HashMap<String, String>();
```

Comment:
This is to keep "Map<String, String> jmsProperties = JMSPropertyMapper.getEnumMap();"
synapse jms property name's against maching ballerina jms keys. This is needed after splitting the JMS uri into key value pairs, to get the matching ballerina jms key that is mapped to relevant synapse jms property. 
Please see the logic written inside "visit(AddressEndpoint addressEndpoint)"